src/GithubFilesBundle/Service/GithubImport.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace GithubFilesBundle\Service;
+
+use CftfBundle\Entity\LsAssociation;
+use CftfBundle\Entity\LsDefItemType;
+use CftfBundle\Entity\LsDefSubject;
+use CftfBundle\Entity\LsDoc;
+use CftfBundle\Entity\LsItem;
+use Doctrine\ORM\EntityManager;
+use JMS\DiExtraBundle\Annotation as DI;
+use Ramsey\Uuid\Uuid;
+
+/**
+ * Class GithubImport.
+ *
+ * @DI\Service("cftf_import.github")
+ */
+class GithubImport
+{
+
+    /**
+     * @var EntityManager
+     */
+    private $em;
+
+    /**
+     * @param EntityManager $em
+     *
+     * @DI\InjectParams({
+     *     "em" = @DI\Inject("doctrine.orm.entity_manager"),
+     * })
+     */
+    public function __construct(EntityManager $em)
+    {
+        $this->em = $em;
+    }
+
+    /**
+     * @return EntityManager
+     */
+    protected function getEntityManager()
+    {
+        return $this->em;
+    }
+
+    /**
+     * Parse an Github document into a LsDoc/LsItem hierarchy
+     *
+     * @param array $lsdocKeys
+     * @param string $fileContent
+     */
+    public function parseGithubDocument($lsDocKeys, $lsItemKeys, $fileContent){
+        $csvContent = str_getcsv($fileContent, "\n");
+        $headers = array();
+        $content = array();
+        $tempContent = array();
+
+        foreach ($csvContent as $i => $row) {
+            $row = str_getcsv($row, ",");
+
+            if ($i == 0){
+                $headers = $row;
+                continue;
+            }
+
+            foreach($headers as $h => $col){
+                $tempContent[$col] = $row[$h];
+            }
+
+            array_push($content, $tempContent);
+            $tempContent = array();
+        }
+
+        $this->saveGithubDocument($lsDocKeys, $lsItemKeys, $content);
+    }
+
+    /**
+     * Save an Github document into a LsDoc/LsItem hierarchy
+     *
+     * @param array $lsdocKeys
+     * @param string $content
+     */
+    public function saveGithubDocument($lsDocKeys, $lsItemKeys, $content){
+        $em = $this->getEntityManager();
+        $lsDoc = new LsDoc();
+
+        foreach($content as $i => $row){
+            if($i < 1){
```

Comment:
The section in this if block should probably go before the foreach to avoid having to do the if check for every line since it is always done once per loaded file.